shared/src/main/scala/squants/Dimensionless.scala

```diff
@@ -47,7 +47,7 @@ final class Dimensionless private (val value: Double, val unit: DimensionlessUni
  */
 object Dimensionless extends Dimension[Dimensionless] {
   def apply[A](n: A, unit: DimensionlessUnit)(implicit num: Numeric[A]) = new Dimensionless(num.toDouble(n), unit)
-  def apply = parse _
+  def apply(value: Any) = parse(value)
```

Comment:
Is this change required by dotty?

Reply:
Yes. See this [comment](https://github.com/lampepfl/dotty/issues/2994#issuecomment-326533372)

Reply:
👍